### PR TITLE
x/oracle: generic oracle with initial uniswap data implementation

### PR DESCRIPTION
**Pull request by jackzampolin**
This PR is now in working state. To run perform the following commands from repo root on this branch:

```bash
# start a chain and configure the oracle-feeder app to work with this new chain
./dev-env start

# in another window you can query using the oracle-feeder
oracle-feeder q oracle-data
```

Last item remaining is the querying of the stored oracle data.

### CODE REVIEW

**Comment by jackzampolin** [comment]
Ok, so new plan here:

- [x] Proto files
- [x] Initial keeper code to support msgs and queries
- [x] Msg and query handlers
- [ ] feeder client
  * [x] Cmd
  * [ ] keys (WIP)
  * [ ] config (grpc endpoint and 
  * [ ] querying data
  * [ ] sending txs
  * [ ] transaction loop
- [ ] Running App
- [ ] Shell script to spin up single validator app w/ feeder pulling data
- [ ] Voting periods implementation
- [ ] Begin/endblock implementations

**Comment by fedekunze** [comment]
jack do you want to fix conflicts so that I can review? 🙏

File: x/oracle/abci.go [hunk 1]
```diff
@@ -120,11 +121,27 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
...
-				k.StakingKeeper.Slash(ctx, cons, ctx.BlockHeight(), sval.GetConsensusPower(), params.SlashFraction)
+			missedVotesPerWindow := sdk.NewDec(counter).Quo(sdk.NewDec(params.SlashWindow))
+			if params.MinValidPerWindow.GTE(missedVotesPerWindow) {
+				// continue with next counter
+				return false
+			}
+
+			// slash validator below the minimum vote threshold
+			validator := k.StakingKeeper.Validator(ctx, sdk.ValAddress(val))
+			if validator == nil {
```
Comment (jackzampolin) [inline x/oracle/abci.go]: This condition should never be hit.

File: x/oracle/genesis.go [hunk 2]
```diff
@@ -8,15 +8,26 @@ import (
 
 // InitGenesis initialize default parameters
 // and the keeper's address to pubkey map
-func InitGenesis(ctx sdk.Context, k keeper.Keeper, data types.GenesisState) {
-	k.SetParams(ctx, data.Params)
+func InitGenesis(ctx sdk.Context, k keeper.Keeper, gs types.GenesisState) {
+	k.SetParams(ctx, gs.Params)
+	// Set the vote period at initialization
+	k.SetVotePeriodStart(ctx, ctx.BlockHeight())
```
Comment (jackzampolin) [inline x/oracle/genesis.go:14]: this is much better lol

**Pull request merged**